core/src/autogluon/core/scheduler/hyperband.py

```diff
@@ -315,7 +317,7 @@ def __init__(self, train_fn, **kwargs):
             dict_name='scheduler_options')
         resume = kwargs['resume']
         scheduler_type = kwargs['type']
-        supported_types = {'stopping', 'promotion'}
```

Comment:
I don't like this. Please try not to insert lots of code into HyperbandScheduler that only makes sense for a subclass. If you cannot do this by formulating some private methods, to be coded in the subclass, then better use copy&paste for the constructor.

We also don't want includes to subclass code in HyperbandScheduler.